content/5_Plat/Convex_Hull.mdx

```diff
@@ -85,22 +84,205 @@ export const problems = {
 
 <!-- [convex hull](https://en.wikipedia.org/wiki/Convex_hull_algorithms). -->
 
+
+## Introduction
+
+The **Convex Hull** is the subset of points that forms the smallest convex polygon which encloses all points in the set.
+To visualize this, imagine that each point is a pole. Then, imagine what happens if you were to wrap a rope around the outside of all the poles, and then pull infinitely hard, such that the connections between any two points that lie on the edge of the rope are lines. The set of points that touch the rope (and are not colinear) is the convex hull.
+
+<Spoiler title="Convex Hull Visualization">
+
+![Convex Hull Visualization](./assets/convexhull.jpg)
+
+</Spoiler>
+
 <FocusProblem problem={problems.sample[0]} />
 
 ## With Graham Scan
 
-- [cp-algo](https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html)
-- [My Implementation](<https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h>)
+<Resources>
+	<Resource source="cp-algo" title="Graham Scan" url="https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html" starred />
+	<Resource source="Benq" title="Graham Scan Implementation" url="https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h" />
+	<Resource source="VisuAlgo" title="Graham Scan Visualization" url="https://visualgo.net/en/convexhull" />
+	<Resource source="UCSD" title="Original Graham Scan Paper" url="http://www.math.ucsd.edu/~ronspubs/72_10_convex_hull.pdf" />
+</Resources>
+
+<Youtube id="9rQMLpQn5xQ"/>
+
+## Tutorial
+
+The Graham Scan algorithm works in 3 steps. First, it sorts all of the $n$ points by their counterclockwise angle around an abitrary pivot $P_0$. This algorithm uses the bottommost (and rightmost if there is a tie), point as $P_0$. 
+
+To start the creation of the convex hull, we choose 3 points. The pivot, the first, and the last point based on our initial sorting.
```

Comment:
huh, isn't the pivot the first point

Reply:
still hasn't been updated

Reply:
> The pivot (first point), the second point, and the last point based on our initial sorting.

 is what I see.

Reply:
Er it changed, but the last point isn't included in the stack at the start. Also, it's leftmost, not rightmost.

Reply:
huh, I was reading through cp4 and it mentioned it but since we are using your implementation i changed it to match (also wouldn't it not change the hull regarding left or right as long as its an extreme)